### PR TITLE
Add support for announcing services

### PR DESCRIPTION
**Pull request by alesharik**
This PR is a fork of #10 with fixed CI build, updated dependencies, warning and deprecation cleanup

### CODE REVIEW

**Comment by alesharik** [comment]
@azasypkin any updates on merging this?

File: src/adapters/avahi/adapter.rs [hunk 1]
```diff
@@ -445,6 +445,62 @@ impl HostAdapter for AvahiAdapter {
...
+            let error = AvahiError::from_error_code(code);
+            return Err(From::from(error));
+        }
+
+        let service_name = AvahiUtils::to_c_string(service_name.to_owned()).into_raw();
+        let service_type = AvahiUtils::to_c_string(service_type.to_owned()).into_raw();
+        let result_code = unsafe {
+            avahi_entry_group_add_service(
+                entry_group,
+                -1,
```
Comment (azasypkin) [inline src/adapters/avahi/adapter.rs]: __nit:__
```suggestion
                AvahiIfIndex::AVAHI_IF_UNSPEC,
```

File: src/bindings/avahi/functions.rs [hunk 2]
```diff
@@ -300,6 +300,19 @@ extern "C" {
         size: usize,
     ) -> c_int;
 
+    pub fn avahi_entry_group_add_service(
+        group: *mut AvahiEntryGroup,
+        interface: c_int,
```
Comment (azasypkin) [inline src/bindings/avahi/functions.rs]: __nit:__
```suggestion
        interface: AvahiIfIndex,
```

**Review by azasypkin — CHANGES_REQUESTED** [review]
Sorry for the delay, I just don't actively maintain/use this repo anymore. The change looks reasonable, just two tiny nits before we can merge it and publish new version.

The code base is very old and inefficient one, and obviously needs some love :slightly_smiling_face: 

Thanks for the contribution!

**Review by azasypkin — APPROVED** [review]
LGTM

**Comment by andir** [comment]
Thanks!

**Pull request merged**